jib-core/src/main/java/com/google/cloud/tools/jib/image/HistoryEntry.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.image;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/** Represents an item in the container configuration's {@code history} list. */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class HistoryEntry implements JsonTemplate {
```

Comment:
I think this might better belong under the `image.json` package, next to `ContainerConfigurationTemplate`.

I'm thinking it might make more sense to call this `ContainerConfigurationHistory` and have it manage the actual JSON templates. Then, `Image` will have a `ContainerConfigurationHistory` and not have to directly manage `HistoryEntry` JSON templates.

Reply:
I'll move it to the package, but I'm just going to make an issue for the second point for now.

Reply:
#889